octavia-cli/octavia_cli/generate/renderer.py

```diff
@@ -0,0 +1,161 @@
+            return self.field_metadata.get(name)
+
+    @property
+    def _is_array_of_objects(self) -> bool:
+        if self.type == "array" and self.items:
+            if self.items["type"] == "object":
+                return True
+        return False
+
+    def _get_one_of_values(self) -> List[List["FieldToRender"]]:
+        """An object field get have multiple kind of fields if it's a oneOf.
+        This functions returns all the possible one of values the field can take.
+        Returns:
+            [list]: List of oneof values.
+        """
+        if not self.oneOf:
+            return []
+        one_of_values = []
+        for one_of_value in self.oneOf:
+            properties = get_object_fields(one_of_value)
+            one_of_values.append(properties)
+        return one_of_values
+
+    def _get_array_items(self) -> List["FieldToRender"]:
+        """If the field is an array of objects, retrieve fields of these objects.
+
+        Returns:
+            [list]: List of fields
+        """
+        if self._is_array_of_objects:
+            required_fields = self.items.get("required", [])
+            return parse_fields(required_fields, self.items["properties"])
+        return []
+
+    def _get_required_comment(self) -> str:
+        return "REQUIRED" if self.required else "OPTIONAL"
+
+    def _get_type_comment(self) -> str:
+        return self.type if self.type else None
+
+    def _get_secret_comment(self) -> str:
+        return "🤫" if self.airbyte_secret else None
+
+    def _get_description_comment(self) -> str:
+        return self.description if self.description else None
+
+    def _get_example_comment(self) -> str:
+        example_comment = None
+        if self.examples:
+            if isinstance(self.examples, list):
+                if len(self.examples) > 1:
+                    example_comment = f"Examples: {', '.join([str(example) for example in self.examples])}"
+                else:
+                    example_comment = f"Example: {self.examples[0]}"
+            else:
+                example_comment = f"Example: {self.examples}"
+        return example_comment
+
+    def _get_default(self) -> str:
+        if self.const:
+            return self.const
+        return self.default
+
+    @staticmethod
+    def _build_comment(comment_functions: Callable) -> str:
+        return " | ".join(filter(None, [comment_fn() for comment_fn in comment_functions])).replace("\n", "")
+
+
+def parse_fields(required_fields: List[str], fields: dict) -> List["FieldToRender"]:
+    return [FieldToRender(f_name, f_name in required_fields, f_metadata) for f_name, f_metadata in fields.items()]
+
+
+def get_object_fields(field_metadata: dict) -> List["FieldToRender"]:
+    if field_metadata.get("properties"):
+        required_fields = field_metadata.get("required", [])
+        return parse_fields(required_fields, field_metadata["properties"])
+    return []
+
+
+class ConnectionSpecificationRenderer:
```

Comment:
Should this be `ConnectorSpecificationRenderer` rather than `ConnectionSpecificationRenderer`?

Reply:
In the [API response from Airbyte](https://airbyte-public-api-docs.s3.us-east-2.amazonaws.com/rapidoc-api-docs.html#post-/v1/source_definition_specifications/get) the field is named `connectionSpecification` this I why I chosed this name.